fetch/handler.go

```diff
@@ -60,23 +64,27 @@ func (h *handler) handleEpochInfoReq(ctx context.Context, msg []byte) ([]byte, e
 	if err := codec.Decode(msg, &epoch); err != nil {
 		return nil, err
 	}
-	atxids, err := atxs.GetIDsByEpoch(h.cdb, epoch)
-	if err != nil {
-		h.logger.WithContext(ctx).With().Warning("serve: failed to get epoch atx IDs", epoch, log.Err(err))
-		return nil, err
-	}
-	ed := EpochData{
-		AtxIDs: atxids,
-	}
-	h.logger.WithContext(ctx).With().Debug("serve: responded to epoch info request",
-		epoch,
-		log.Int("atx_count", len(ed.AtxIDs)),
-	)
-	bts, err := codec.Encode(&ed)
-	if err != nil {
-		h.logger.WithContext(ctx).With().Fatal("serve: failed to serialize epoch atx", epoch, log.Err(err))
-	}
-	return bts, nil
+
+	cacheKey := sql.QueryCacheKey(atxs.CacheKindEpochATXs, epoch.String())
+	return sql.WithCachedSubKey(h.cdb, cacheKey, fetchSubKey, func() ([]byte, error) {
+		atxids, err := atxs.GetIDsByEpoch(h.cdb, epoch)
+		if err != nil {
+			h.logger.WithContext(ctx).With().Warning("serve: failed to get epoch atx IDs", epoch, log.Err(err))
+			return nil, err
+		}
+		ed := EpochData{
+			AtxIDs: atxids,
+		}
+		h.logger.WithContext(ctx).With().Debug("serve: responded to epoch info request",
+			epoch,
+			log.Int("atx_count", len(ed.AtxIDs)),
+		)
```

Comment:
btw WithContext(ctx) and With() allocates a lot even if level is disabled, might be a problem if it will be on the hot path. we agreed to switch to regular zap, and there is also a helper log.Context(ctx) that will not execute if level is disabled.